pkg/audit/verifier.go

```diff
@@ -487,23 +511,46 @@ func (verifier *Verifier) GetShare(ctx context.Context, limit *pb.AddressedOrder
 	}, nil
 }
 
+// RemoveFailedPieces removes lost pieces from a pointer
+func (verifier *Verifier) RemoveFailedPieces(ctx context.Context, path string, pointer *pb.Pointer, failedNodes storj.NodeIDList) (err error) {
+	defer mon.Task()(&ctx)(&err)
+	if len(failedNodes) == 0 {
+		return nil
+	}
+	remoteSegment := pointer.GetRemote()
+	newRemotePieces := remoteSegment.RemotePieces[:0]
```

Comment:
why have `remoteSegment.RemotePieces[:0]` ?

Reply:
To save allocations, it reuses the same slice

Reply:
In my opinion , modifying passed parameters should be avoided as much as possible, however you need to do so for avoiding copies, here it looks that's needed.